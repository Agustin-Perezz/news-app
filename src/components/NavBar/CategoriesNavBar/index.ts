export * from './CategoriesNavBar';
export * from './listCategories';
