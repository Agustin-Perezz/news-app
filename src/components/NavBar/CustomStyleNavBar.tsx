import { styled } from '@mui/material/styles';
import MuiBox from '@mui/material/Box';

export const CustomBoxContainer = styled(MuiBox)(() => ({
  backgroundColor: '#292929',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  height: 'max-content',
  flexWrap: 'wrap',
  paddingTop: '20px',
}));

export const CustomNavBarBox = styled(MuiBox)(() => ({
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  width: '600px',
  margin: '0px 0px 20px 0px',
  '@media (min-width: 1003px)': {
    width: '880px',
    justifyContent: 'space-around',
  },
}));
