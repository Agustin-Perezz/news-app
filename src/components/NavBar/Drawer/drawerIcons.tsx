import AllOutIcon from '@mui/icons-material/AllOut';
import HomeIcon from '@mui/icons-material/Home';
import DashboardIcon from '@mui/icons-material/Dashboard';
import ScienceIcon from '@mui/icons-material/Science';
import SportsFootballIcon from '@mui/icons-material/SportsFootball';
import PaidIcon from '@mui/icons-material/Paid';
import LocalHospitalIcon from '@mui/icons-material/LocalHospital';
import SportsEsportsIcon from '@mui/icons-material/SportsEsports';
import DeveloperBoardIcon from '@mui/icons-material/DeveloperBoard';
import AutoStoriesIcon from '@mui/icons-material/AutoStories';
import IcecreamIcon from '@mui/icons-material/Icecream';
import ConnectingAirportsIcon from '@mui/icons-material/ConnectingAirports';

export const drawerIcons = [
  { icon: <AllOutIcon /> },
  { icon: <HomeIcon /> },
  { icon: <DashboardIcon /> },
  { icon: <ScienceIcon /> },
  { icon: <SportsFootballIcon /> },
  { icon: <PaidIcon /> },
  { icon: <LocalHospitalIcon /> },
  { icon: <SportsEsportsIcon /> },
  { icon: <DeveloperBoardIcon /> },
  { icon: <AutoStoriesIcon /> },
  { icon: <IcecreamIcon /> },
  { icon: <ConnectingAirportsIcon /> },
];
