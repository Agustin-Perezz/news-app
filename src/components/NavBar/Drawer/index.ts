export * from './DrawerCategories';
export * from './DrawerCategory';
export * from './NavBarDrawer';
export * from './drawerIcons';
