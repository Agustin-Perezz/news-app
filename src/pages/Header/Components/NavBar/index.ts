export * from './NavBar';
export * from './CustomStylesNavBar';
