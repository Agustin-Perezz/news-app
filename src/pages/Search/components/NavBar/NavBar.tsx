import { useContext } from 'react';
import { ImageListItem, Typography } from '@mui/material';
import { CustomStack } from '../../../../components/CustomStack';
import { SocialMedia } from '../../../../components/ui/SocialMedia';
import { SearchBar } from '../SearchBar';
import { NewsContext } from '../../../../context/NewsContext';

import news_logo from '../../../../assets/images/news-logo.png';

export const NavBar: React.FC = () => {
  return (
    <CustomStack
      stackProps={{
        direction: 'row',
        justifyContent: 'space-evenly',
        alignItems: 'center',
        marginBottom: -1.5,
      }}
    >
      <ImageListItem sx={{ width: 190 }}>
        <img src={news_logo} alt="news-logo" />
      </ImageListItem>
      <SearchBar />
      <SocialMedia />
    </CustomStack>
  );
};
