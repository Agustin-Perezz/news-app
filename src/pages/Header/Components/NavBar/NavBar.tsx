import Box from '@mui/material/Box';
import { ImageListItem } from '@mui/material';
import { SearchBar } from '../SearchBar';
import { CategoriesNavBar } from '../CategoriesNavBar';
import { CustomStack } from '../../../../components/CustomStack';
import { SocialMedia } from '../../../../components/ui/SocialMedia';
import { useSearch } from '../../hooks';

import news_logo from '../../../../assets/images/news-logo.png';

export const NavBar: React.FC = () => {
  const { handleSubmit } = useSearch();
  return (
    <Box
      sx={{
        backgroundColor: '#292929',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        height: 'max-content',
        flexWrap: 'wrap',
        paddingTop: '20px',
      }}
    >
      <CustomStack
        stackProps={{
          direction: 'row',
          justifyContent: 'center',
          alignItems: 'center',
          width: '1000px',
        }}
      >
        <ImageListItem sx={{ width: 150 }}>
          <img src={news_logo} alt="news-logo" />
        </ImageListItem>
        <SearchBar handleSubmit={handleSubmit} />
        <SocialMedia />
      </CustomStack>
      <CategoriesNavBar handleClick={handleSubmit} />
    </Box>
  );
};
