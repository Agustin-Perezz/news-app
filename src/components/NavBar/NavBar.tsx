import { ImageListItem } from '@mui/material';
import useMediaQuery from '@mui/material/useMediaQuery';
import { SearchBar } from './SearchBar';
import { SocialMedia } from '../ui/SocialMedia';
import { CategoriesNavBar } from './CategoriesNavBar';
import { NavBarDrawer } from './Drawer';
import { CustomBoxContainer, CustomNavBarBox } from './CustomStyleNavBar';
import { CategoryProvider } from '../../context/CategoryContext';
import { useSearch } from '../../hooks';

import news_logo from '../../assets/images/news-logo.png';

export const NavBar: React.FC = () => {
  const { handleSubmit } = useSearch();
  const matches = useMediaQuery('(max-width:1000px)');

  return (
    <CategoryProvider>
      <CustomBoxContainer>
        <CustomNavBarBox>
          {matches ? (
            <NavBarDrawer handleSubmit={handleSubmit} />
          ) : (
            <ImageListItem sx={{ maxWidth: '130px' }}>
              <img src={news_logo} alt="news-logo" />
            </ImageListItem>
          )}
          <SearchBar handleSubmit={handleSubmit} />
          {!matches && <SocialMedia />}
        </CustomNavBarBox>
        {!matches && <CategoriesNavBar handleClick={handleSubmit} />}
      </CustomBoxContainer>
    </CategoryProvider>
  );
};
