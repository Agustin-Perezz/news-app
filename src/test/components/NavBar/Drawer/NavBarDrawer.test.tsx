import { fireEvent, screen } from '@testing-library/react';
import { NavBarDrawer } from '../../../../components/NavBar/Drawer';
import { renderWhitContext } from '../../../utils';

describe('Test in <NavBarDrawer />', () => {
  const handleSubmit = jest.fn();
  function setup() {
    return renderWhitContext({
      children: <NavBarDrawer handleSubmit={handleSubmit} />,
      pathname: '/category/home',
      cache: {},
    });
  }

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('should init with hidden drawer', () => {
    setup();
    const icon = screen.getByTestId('MenuRoundedIcon');
    expect(icon).toHaveAttribute('aria-hidden', 'true');
  });
  it('should open the drawer when click button', async () => {
    setup();
    fireEvent.click(screen.getByRole('button'));
    const drawerContent = screen.getByRole('presentation');
    expect(drawerContent).toBeInTheDocument();
  });
  it('should call handleSubmit when click in the any category of the drawer', () => {
    setup();
    fireEvent.click(screen.getByRole('button'));
    fireEvent.click(screen.getByText('sports'));
    expect(handleSubmit).toHaveBeenCalled();
  });
});
