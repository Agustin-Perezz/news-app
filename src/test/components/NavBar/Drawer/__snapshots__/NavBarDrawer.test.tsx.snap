// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <NavBarDrawer /> should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="MenuRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M4 18h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1zm0-5h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1zM3 7c0 .55.45 1 1 1h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
